File: python/기초 문제/10828.py
import sys
n = int(sys.stdin.readline().rstrip())
list = []
for i in range(n):
    order = sys.stdin.readline().rstrip()
    if order[0:4] == "push":
        a = int(order[5:])
        list.append(a)
    elif order == "pop":
        if len(list) == 0:
            print(-1)
        else:
            print(list[-1])
            list.pop()
    elif order == "size":
        print(len(list))
    elif order == "empty":
        if len(list) == 0:
            print(1)
        else:
            print(0)
    elif order == "top":
        if len(list) == 0:
            print(-1)
        else:
            print(list[-1])

File: c++/self_study/10828.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    int N, k;
    int pos = -1;
    string order;
    cin >> N;
    vector<int> stack;
    for(int i = 0; i < N; i++){
        cin >> order;
        if(order == "push"){
            cin >> k;
            stack.push_back(k);
            pos++;
        }
        else if(order == "pop"){
            if(pos == -1){
                cout << -1 << "\n";
            }
            else{
                cout << stack[pos] << "\n";
                stack.erase(stack.begin() + pos);
                pos--;
            }
        }
        else if (order == "size")
        {
            cout << stack.size() << "\n";
        }
        else if (order == "empty"){
            if(pos == -1){
                cout << 1 << "\n";
            }
            else{
                cout << 0 << "\n";
            }
        }
        else if(order == "top"){
            if(pos == -1){
                cout << -1 << "\n";
            }
            else{
                cout << stack[pos] << "\n";\
            }
        }
    }
}